src/main/kotlin/App.kt

```diff
@@ -0,0 +1,177 @@
+                    if (sc == ScreenType.ARCH_CREATE) currentArch.name = name else currentNote.name = name
+                }
+
+                if (isNeedDesc) {
+                    menu[menu.size] = MenuItem("Ввести текст") {
+                        println("Введите текст заметки:")
+                        val desc = input.nextLine()
+                        currentNote.desc = desc
+                    }
+                }
+
+                menu[menu.size] = (MenuItem("Сохранить и выйти") {
+
+                    if (sc == ScreenType.ARCH_CREATE) {
+                        if (currentArch.isReadyToSave()) {
+                            archList.add(currentArch)
+                            changeScreen(selectSc(sc))
+                        }
+                    } else if (sc == ScreenType.NOTE_CREATE)
+                        if (currentNote.isReadyToSave()) {
+                            currentArch.notes.add(currentNote)
+                            changeScreen(selectSc(sc))
+                        }
+
+                })
+
+                menu[menu.size] = MenuItem("Выход\n") {
+
+                    changeScreen(selectSc(sc))
+
+                }
+
+            }
+
+            ScreenType.NOTE_TEXT -> {
+
+                menu[menu.size] = MenuItem("Выход\n") {
+
+                    changeScreen(selectSc(sc))
+
+                }
+
+            }
+
+            else -> {
+
+            }
+        }
+
+    }
+
+    fun showMenu() {
+
+        println(sc.header)
+
+        if (sc == ScreenType.NOTE_TEXT) println(currentNote.toString())
+
+        val sortedMenu = menu.toSortedMap()
+        for (i in sortedMenu) {
+            println("${i.key}: ${i.value}")
+        }
+    }
+
+    fun isMenuKeyCorrect(key: Int) = menu.containsKey(key)
+
+    fun onClick(key: Int) {
+
+        val item = menu[key] ?: return
+        if (item.func != null) {
+
+            item.func?.invoke()
+            return
+
+        }
+
+        when (sc) {
+
+            ScreenType.ARCH_SELECT, ScreenType.NOTE_SELECT -> {
+
+                if (sc == ScreenType.ARCH_SELECT) {
+                    currentArch = archList[key - 1]
+                } else {
+                    currentNote = currentArch.notes[key - 1]
+                }
+
+                changeScreen(selectSc(sc))
+
+            }
+
+            else -> {
+
+            }
+
+        }
+
+    }
+
+    private fun createSc(sc: ScreenType): ScreenType {
+
+        return when (sc) {
+            ScreenType.ARCH_SELECT -> ScreenType.ARCH_CREATE
+            ScreenType.NOTE_SELECT -> ScreenType.NOTE_CREATE
+            else -> {
+                ScreenType.EXIT
+            }
+        }
+    }
+
+    private fun selectSc(sc: ScreenType): ScreenType {
+        return when (sc) {
+            ScreenType.ARCH_CREATE -> ScreenType.ARCH_SELECT
+            ScreenType.NOTE_CREATE -> ScreenType.NOTE_SELECT
+            ScreenType.ARCH_SELECT -> ScreenType.NOTE_SELECT
```

Comment:
```suggestion
            ScreenType.NOTE_CREATE, ScreenType.ARCH_SELECT -> ScreenType.NOTE_SELECT
```
Можно вот так сокращать